ts/features/payments/onboarding/screens/PaymentsOnboardingFeedbackScreen.tsx

```diff
@@ -77,10 +77,28 @@ const PaymentsOnboardingFeedbackScreen = () => {
     const payment_method_selected = availablePaymentMethods?.find(
       paymentMethod => paymentMethod.id === selectedPaymentMethodId
     )?.name;
-    if (outcome === WalletOnboardingOutcomeEnum.SUCCESS) {
-      analytics.trackSuccessOnboardingPaymentMethod({
-        payment_method_selected
-      });
+
+    switch (outcome) {
+      case WalletOnboardingOutcomeEnum.SUCCESS:
+        analytics.trackSuccessOnboardingPaymentMethod({
+          payment_method_selected
+        });
+        break;
+      case WalletOnboardingOutcomeEnum.AUTH_ERROR:
+        analytics.trackOnboardingPaymentMethodDenied({
+          payment_method_selected
+        });
+        break;
+      case WalletOnboardingOutcomeEnum.CANCELED_BY_USER:
+        analytics.trackAddOnboardingPaymentMethodCanceled({
+          payment_method_selected
+        });
+        break;
+      case WalletOnboardingOutcomeEnum.ALREADY_ONBOARDED:
+        analytics.trackAddOnboardingPaymentMethodDuplicated({
+          payment_method_selected
+        });
+        break;
```

Comment:
What do you think about consolidating the error cases into a single function in `/analytics/index.ts`, for example, `trackFailureOnboardingPaymentMethod`? This function could take both `payment_method_selected` and `outcome` as input, and internally map the outcome to the correct event to emit. This would reduce the need for multiple methods, making the code easier to maintain and extend.

Reply:
Sounds good to me! 💪